pwm-imx platform device must now be registered with dev-id (imx1-pwm or
imx27-pwm) to match driver compatible list.

Signed-off-by: Gwenhael Goavec-Merou <gwenhael.goavec-merou@armadeus.com>
---

Index: linux-3.7.6/arch/arm/plat-mxc/devices/platform-mxc_pwm.c
===================================================================
--- linux-3.7.6.orig/arch/arm/plat-mxc/devices/platform-mxc_pwm.c
+++ linux-3.7.6/arch/arm/plat-mxc/devices/platform-mxc_pwm.c
@@ -9,9 +9,10 @@
 #include <mach/hardware.h>
 #include <mach/devices-common.h>
 
-#define imx_mxc_pwm_data_entry_single(soc, _id, _hwid, _size)		\
+#define imx_mxc_pwm_data_entry_single(soc, _devid, _id, _hwid, _size)	\
 	{								\
 		.id = _id,						\
+		.devid = _devid,			\
 		.iobase = soc ## _PWM ## _hwid ## _BASE_ADDR,		\
 		.iosize = _size,					\
 		.irq = soc ## _INT_PWM ## _hwid,			\
@@ -21,31 +22,31 @@
 
 #ifdef CONFIG_SOC_IMX21
 const struct imx_mxc_pwm_data imx21_mxc_pwm_data __initconst =
-	imx_mxc_pwm_data_entry_single(MX21, 0, , SZ_4K);
+	imx_mxc_pwm_data_entry_single(MX21, "imx1-pwm", 0, , SZ_4K);
 #endif /* ifdef CONFIG_SOC_IMX21 */
 
 #ifdef CONFIG_SOC_IMX25
 const struct imx_mxc_pwm_data imx25_mxc_pwm_data[] __initconst = {
-#define imx25_mxc_pwm_data_entry(_id, _hwid)				\
-	imx_mxc_pwm_data_entry(MX25, _id, _hwid, SZ_16K)
-	imx25_mxc_pwm_data_entry(0, 1),
-	imx25_mxc_pwm_data_entry(1, 2),
-	imx25_mxc_pwm_data_entry(2, 3),
-	imx25_mxc_pwm_data_entry(3, 4),
+#define imx25_mxc_pwm_data_entry(_devid, _id, _hwid)			\
+	imx_mxc_pwm_data_entry(MX25, _devid_, _id, _hwid, SZ_16K)
+	imx25_mxc_pwm_data_entry("imx27-pwm", 0, 1),
+	imx25_mxc_pwm_data_entry("imx27-pwm", 1, 2),
+	imx25_mxc_pwm_data_entry("imx27-pwm", 2, 3),
+	imx25_mxc_pwm_data_entry("imx27-pwm", 3, 4),
 };
 #endif /* ifdef CONFIG_SOC_IMX25 */
 
 #ifdef CONFIG_SOC_IMX27
 const struct imx_mxc_pwm_data imx27_mxc_pwm_data __initconst =
-	imx_mxc_pwm_data_entry_single(MX27, 0, , SZ_4K);
+	imx_mxc_pwm_data_entry_single(MX27, "imx27-pwm", 0, , SZ_4K);
 #endif /* ifdef CONFIG_SOC_IMX27 */
 
 #ifdef CONFIG_SOC_IMX51
 const struct imx_mxc_pwm_data imx51_mxc_pwm_data[] __initconst = {
-#define imx51_mxc_pwm_data_entry(_id, _hwid)				\
-	imx_mxc_pwm_data_entry(MX51, _id, _hwid, SZ_16K)
-	imx51_mxc_pwm_data_entry(0, 1),
-	imx51_mxc_pwm_data_entry(1, 2),
+#define imx51_mxc_pwm_data_entry(_devid, _id, _hwid)			\
+	imx_mxc_pwm_data_entry(MX51, _devid, _id, _hwid, SZ_16K)
+	imx51_mxc_pwm_data_entry("imx27-pwm", 0, 1),
+	imx51_mxc_pwm_data_entry("imx27-pwm", 1, 2),
 };
 #endif /* ifdef CONFIG_SOC_IMX51 */
 
@@ -64,6 +65,6 @@ struct platform_device *__init imx_add_m
 		},
 	};
 
-	return imx_add_platform_device("mxc_pwm", data->id,
+	return imx_add_platform_device(data->devid, data->id,
 			res, ARRAY_SIZE(res), NULL, 0);
 }
